scripts/cnn_variant_wdl/cram2model.wdl

```diff
@@ -0,0 +1,431 @@
+    }
+
+    call SplitIntervals {
+        input:
+            scatter_count = scatter_count,
+            intervals = calling_intervals,
+            ref_fasta = reference_fasta,
+            ref_dict = reference_dict,
+            ref_fai = reference_fasta_index,
+            gatk_docker = gatk_docker,
+            gatk_override = gatk_override,
+            preemptible_attempts = preemptible_attempts
+    }
+
+    scatter (calling_interval in SplitIntervals.interval_files) {
+
+        call RunHC4 {
+            input:
+                input_bam = CramToBam.output_bam,
+                input_bam_index = CramToBam.output_bam_index,
+                reference_fasta = reference_fasta,
+                reference_dict = reference_dict,
+                reference_fasta_index = reference_fasta_index,
+                output_prefix = output_prefix,
+                interval_list = calling_interval,
+                gatk_docker = gatk_docker,
+                gatk_override = gatk_override,
+                preemptible_attempts = preemptible_attempts,
+                extra_args = extra_args,
+                disk_space_gb = disk_space_gb
+        }
+
+        call WriteTensors {
+            input:
+                input_vcf = RunHC4.raw_vcf,
+                input_vcf_index = RunHC4.raw_vcf_index,
+                input_bam = RunHC4.bamout,
+                input_bam_index = RunHC4.bamout_index,
+                truth_vcf = truth_vcf,
+                truth_vcf_index = truth_vcf_index,
+                truth_bed = truth_bed,
+                tensor_type = tensor_type,
+                reference_fasta = reference_fasta,
+                reference_dict = reference_dict,
+                reference_fasta_index = reference_fasta_index,
+                output_prefix = output_prefix,
+                interval_list = calling_interval,
+                gatk_docker = gatk_docker,
+                gatk_override = gatk_override,
+                preemptible_attempts = preemptible_attempts,
+                disk_space_gb = disk_space_gb
+        }
+    }
+
+    call MergeVCFs as MergeVCF_HC4 {
+        input:
+            input_vcfs = RunHC4.raw_vcf,
+            output_prefix = output_prefix,
+            gatk_override = gatk_override,
+            gatk_docker = gatk_docker,
+            preemptible_attempts = preemptible_attempts,
+            disk_space = disk_space_gb
+
+    }
+
+    call SamtoolsMergeBAMs {
+        input:
+            input_bams = RunHC4.bamout,
+            output_prefix = output_prefix
+    }
+
+    call TrainModel {
+        input:
+            tar_tensors = WriteTensors.tensors,
+            output_prefix = output_prefix,
+            tensor_type = tensor_type,
+            gatk_override = gatk_override,
+            disk_space_gb = disk_space_gb,
+            epochs = epochs
+    }
+
+    output {
+        MergeVCF_HC4.*
+        SamtoolsMergeBAMs.*
+        TrainModel.*
+    }
+
+}
+
+task CramToBam {
```

Comment:
Yeah, this is screaming to be made a shared task.

Reply:
I was afraid to do this because of firecloud, but I guess it is safe now?

Reply:
It can be done.  It's not pretty, but Firecloud does support subworkflows.